examples/transactions/transactional_consumer.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+func startConsuming(ctx context.Context, kafkaBrokers string, topic string) {
+	client, err := kgo.NewClient(
+		kgo.SeedBrokers(strings.Split(kafkaBrokers, ",")...),
+		// Only read messages that have been written as part of committed transactions.
+		kgo.FetchIsolationLevel(kgo.ReadCommitted()),
+	)
+	if err != nil {
+		fmt.Printf("error initializing Kafka consumer: %v\n", err)
+		return
+	}
+
+	client.AssignGroupTransactSession(
```

Comment:
A group transaction session is generally meant for a consumer-modify-produce type of loop, but the below code does not produce back its modifications. When assigning a transact session, you'd definitely want to use the `GroupTransactSession` return to ensure that things are done properly (commit vs. abort, stuff like that).

In this case, you could just AssignGroup.

Reply:
(if you do want to use a GroupTransactSession, an example is here: https://github.com/twmb/kcl/blob/master/commands/transact/transact.go#L281-L307)